### ISSUE TITLE
x/build: start using Gerrit hashtags for managing open CLs

### ISSUE THREAD

**Issue by bradfitz** [golang/go#24836]
In chat with @andybons @FiloSottile @bcmills and @ianlancetaylor yesterday, I proposed we start using Gerrit's "hashtags" support, now that our hosted Gerrit supports it (notedb support is active for us) and PolyGerrit (the new web UI) supports it.

After discussion, we settled on using tags:

```
so, how about: "wait-author", "wait-release", "wait-issue", "wait-$PERSON"
where $PERSON is some substring of a name/email in the reviewer set.
and if somebody is named "Issue Release" and their email is issue@release.com, then you have to use their full email in $PERSON
we can mass remove the "wait-release" labels when tree opens
```

While triaging today, I found we should also add support for:

```
wait-cl-NNNNN -- this CL is blocked until NNN is resolved
```

I started looking into adding hashtag support to maintner and gopherbot.

Tracking that work here, then documenting all this on the wiki.

**Comment by FiloSottile** [comment]
Also `wait-issue-NNNNN`. I guess the number can be implicit when the issue is linked from the commit message.

**Comment by bradfitz** [comment]
@FiloSottile, yup, I just found that too and just tagged an issue with that, for a CL that had an open proposal but wasn't mentioned in the commit message.

**Comment by bcmills** [comment]
I think I must be missing something, but how do I add hashtags from the Gerrit UI?

**Comment by bradfitz** [comment]
@bcmills, left bar of the PolyGerrit UI:
<img width="256" alt="screen shot 2018-04-16 at 3 56 12 pm" src="https://user-images.githubusercontent.com/2621/38839277-c162a978-418e-11e8-8da5-550e0593b05d.png">

**Comment by bradfitz** [comment]
Or for the old UI, that little icon on the right:
<img width="450" alt="screen shot 2018-04-16 at 3 57 11 pm" src="https://user-images.githubusercontent.com/2621/38839309-dc76e36e-418e-11e8-9dc0-8d9cd0f73d8d.png">

**Comment by bcmills** [comment]
I see the `ADD HASHTAG` button on my own changes, but not on other changes. Is that expected?

![screenshot 2018-04-17 at 11 05 57](https://user-images.githubusercontent.com/5200974/38878629-631da77c-422f-11e8-9ced-c4a07e9cb3ae.png)

**Comment by bradfitz** [comment]
Yes. We haven't modified the default permissions yet to give our Approvers group edit access.

**Comment by bradfitz** [comment]
Sent change modifying our Gerrit config in https://go-review.googlesource.com/c/All-Projects/+/107556 

```
[access "refs/heads/*"]
...
	editTopicName = group approvers
	label-Run-TryBot = +0..+1 group approvers
	label-Run-TryBot = +0..+1 group may-start-trybots
	label-TryBot-Result = -1..+1 group trybot-result-changers
	editHashtags = group approvers
```

... adding that last line.

**Comment by ALTree** [comment]
@bradfitz is there some note/doc explaining how these "Gerrit hashtags" are used in the Go project? I searched the wiki for "hashtag" but I couldn't find anything. 

I've seen people tagging CLs (mostly with `wait-release`) but I still don't understand what exactly means to do that, what's the full list of the available hashtags, how hashtags are supposed to be used, how they integrate with the infrastructure and the dashboards, and whether `wait-release` supersedes the old R=go1.12  syntax.

**Comment by bcmills** [comment]
I don't see any associated CLs to remove `wait-issue` tags.

I think we should remove `wait-issue-NNNNN` when issue NNNNN either is closed or receives a `NeedsFix` or `Proposal-Accepted` label.

**Comment by bradfitz** [comment]
I don't think we need or want `wait-issue-NNNN`. I think it should only be implicit. That also means I can add `wait-issue` on CLs that add code or API with an associated issue.

The `wait-issue` would only be removed once an issue is referenced and NeedsFix-ed or Proposal-Accepted.

**Comment by bradfitz** [comment]
@ALTree, yes, `wait-release` replaces the old R=go1.12 syntax.

This indeed all needs documentation.

**Comment by dmitshur** [comment]
In [CL 216199](https://golang.org/cl/216199), I've used `wait-time-20200128` to indicate the CL is waiting for the day January 28, 2020 before it can be submitted.
